providers/system_proxy.rb

```diff
@@ -91,10 +92,14 @@
                 fe = Chef::Util::FileEdit.new("/etc/environment")
                 fe.search_file_replace_line(/HTTP_PROXY/i,"HTTP_PROXY=\"#{global_settings['http_proxy']}:#{global_settings['http_proxy_port']}\"")
                 fe.search_file_replace_line(/HTTPS_PROXY/i,"HTTPS_PROXY=\"#{global_settings['https_proxy']}:#{global_settings['https_proxy_port']}\"")
+                fe.write_file
```

Comment:
I am afraid is a bug in Chef 11 code. This method has changed in Chef 12 and (probably) will work as expected in a next GECOS version.

I will accept this PR as is and mark it for a future review.